examples/sort-axis.rs

```diff
@@ -143,5 +150,42 @@ fn main() {
     let c = strings.permute_axis(Axis(1), &perm);
     println!("{:?}", c);
 }
+
 #[cfg(not(feature = "std"))]
 fn main() {}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    #[test]
+    fn test_short_permute_axis() {
+        let a = array![
+            [107998.96484375, -0.00002884350434015374],
+            [107999.0869140625, -0.000029842203725824852],
+            [107999.208984375, -0.00002822303686196206],
+            [107999.3310546875, -0.00001991663949734503],
+            [107999.453125, -9.418284272351735e-6],
+            [107999.5751953125, -6.767519609978144e-7],
+            [107999.697265625, 4.659870092811328e-6],
+            [107999.8193359375, 6.502730803219833e-6],
+            [107999.94140625, 5.7282288448929155e-6],
+            [75600.0927734375, 0.00011850080901105835],
+            [75600.21484375, 0.00010000624409032101],
+            [75600.3369140625, 0.00007447255631007994],
+            [75600.458984375, 0.00004374743024472571],
+            [75600.5810546875, 0.000010122515835114419],
+            [75600.703125, -0.000024209462534106906],
+            [75600.8251953125, -0.000057581837088067716],
+            [75600.947265625, -0.00008825070702257629],
+            [75601.0693359375, -0.00009862067207474166],
+        ];
+
+        let perm = a.sort_axis_by(Axis(0), |i, j| a[[i, 0]] < a[[j, 0]]);
+        let b = a.permute_axis(Axis(0), &perm);
+        assert_eq!(b.row(0), array![75600.0927734375, 0.00011850080901105835]);
```

Comment:
sorry for the nitpick but sorting is best by asserting that every element is <= its predecessor (and some way of checking the other elements in the same row are moved correctly).

Reply:
Agreed!  I'll do this and then squash them up.